### ISSUE TITLE
Legality of the use of Finnish NLS rasters

### ISSUE THREAD

**Issue by ilmaisin** [osmlab/editor-layer-index#769]
At least [this file](/osmlab/editor-layer-index/blob/gh-pages/sources/europe/fi/mml-tausta.geojson) mentions [this forum message](https://forum.openstreetmap.org/viewtopic.php?pid=356296#p356296) as the source of the licensing data. The message contains a quotation from NLS about their position on the OSM licensing.

IMO there are two problems in that:
- The license of the dataset has been changed. [The OSM licensing page](https://www.openstreetmap.org/copyright) still has the old license. The current licensing data is [here](https://www.maanmittauslaitos.fi/en/opendata-licence-cc40). The NLS open data is now licensed as cc-by 4.0. What makes things little more complicated is that NLS wants the attribution to contain the dataset used as well as the date when the dataset was received from NLS.
- OSM does not seem to conform to the old license, either. The copyright page mentioned above does not contain the importing years that should have been included according to the quotation posted at the forum.

I have also opened a [thread](https://forum.openstreetmap.org/viewtopic.php?id=68483) about the license change in the Finnish OSM forum.

**Comment by grischard** [comment]
Thank you for your careful eye.

[The PSI directive transposition](https://eur-lex.europa.eu/legal-content/FR/NIM/?uri=CELEX:32013L0037) would be the relevant law. It shouldn't allow more restrictions on reuse than necessary - they would have to show that showing the date, or attribution at all, is necessary and not just nice to have.

**Comment by ilmaisin** [comment]
> Thank you for your careful eye.
> 
> [The PSI directive transposition](https://eur-lex.europa.eu/legal-content/FR/NIM/?uri=CELEX:32013L0037) would be the relevant law. It shouldn't allow more restrictions on reuse than necessary - they would have to show that showing the date, or attribution at all, is necessary and not just nice to have.

It seems to be a directive, not a law. Directives only direct the national legislation, they are not in itself applicable in court. I have understood that OSM wants to develop an untainted dataset instead of relying on untested lay interpertations of law.

That being said, the forum message strongly suggests that NLS has in principle nothing against the importing of their data to OSM. Maybe I should contact them and ask what they want the /copyright page to read.

**Comment by grischard** [comment]
Directives are indeed transcribed by the member states; the page I linked to, if you scroll down, has the Finnish transcription.

Contacting them like you're suggesting is a good idea, yes!

**Comment by ilmaisin** [comment]
Absurd legal theories are not going to help here. You cannot back up your claims by shoving me a bunch of legislations that are written in a language that you do not even understand.
But anyway, I got a reply from NLS and they would be OK with this if the /copyright page is updated to the form I proposed. 
I have submitted a [PR](https://github.com/openstreetmap/openstreetmap-website/pull/2580). Let's see if OSM's powers-that-be want to merge it.

**Comment by rbuffat** [comment]
@grischard @ilmaisin Is this still an open issue or can this issue be closed?

**Comment by simonpoole** [comment]
AS this has turned up here https://github.com/osmlab/editor-layer-index/issues/769 again. Just a last try to clarify, there are two distinct cases here:

- use of CC BY imagery for tracing: the OSMF has typically for good form asked for a waiver of any rights in the traced data, but even if that isn't forthcoming it would be a bit of an uphill battle for the source to claim infringement (note this is limited to actual  imagery, lidar data and similar, not to images of data). With other words this is in general unproblematic.

- direct use (import) of CC BY 4.0 data. Without the waiver importing such data infects the whole OSM database (aka it would have to be distributed on CC BY 4.0 terms) and is a IMHO non-starter.

Simon

**Comment by ilmaisin** [comment]
> @grischard @ilmaisin Is this still an open issue or can this issue be closed?

Yes, this is still pretty much an open issue, because for some reasons OSM does not want to attribute NLS-FI properly, choosing to stubbornly stick into obsolete attributions instead.

**Comment by ilmaisin** [comment]
@simonpoole So the maps (i. e. [mml-tausta](https://github.com/osmlab/editor-layer-index/blob/gh-pages/sources/europe/fi/mml-tausta.geojson) and [mml-topo](https://github.com/osmlab/editor-layer-index/blob/gh-pages/sources/europe/fi/mml-topo.geojson)) have to be removed, but ortophoto layers can stay?

**Comment by simonpoole** [comment]
@ilmaisin I have literally no say in this.

I would however suggest that the attribution issue is a red herring. The text on https://www.openstreetmap.org/copyright should be changed to reflect that the use ended under the NLSFI licence when they changed to CC BY 4.0 (the data used under that licence hasn't suddenly vanished so the entry needs to remain), this also marks the end of the bespoke agreement between NLSFI and OSM. Going forward any required attribution should be made on the corresponding wiki-page. as for essentially all other sources. 

There is enough precedent that we can safely assume that tracing and copying from maps can only happen if permission has been granted, so IMHO any maps cannot be used for OSM purposes as long as the situation with the waivers has not been resolved. But see my opening statement.

**Comment by jimboolio** [comment]
This is an old issue, but I feel like the issue was not resolved. And the question surfaced to my interest due to the NLS data being broken due to changed source configurations. However, if the legality of that data is questionable, then it is not worth fixing unless agreed as legal to continue to do so.

### Issue: 
Current NLS data is CC-BY 4.0; OSM does not have a waiver in this case, or I couldn't find one

As I understood, old NLS data is okay to use (licensed under: https://www.maanmittauslaitos.fi/avoindata-lisenssi-versio1)

#### Current NLS data licensed under CC-BY 4.0 is in my understanding configured as default background layer for iD in most of Finland. Whether that is okay, I tried to look at it (tldr; I did not find proper permission):

> Thank you for your careful eye.
> 
> [The PSI directive transposition](https://eur-lex.europa.eu/legal-content/FR/NIM/?uri=CELEX:32013L0037) would be the relevant law. It shouldn't allow more restrictions on reuse than necessary - they would have to show that showing the date, or attribution at all, is necessary and not just nice to have.

If I understood this correctly, this reply was about ways of proper attribution and not necessarily about the usage of the NLS CC-BY 4.0 data?

---

My investigation on the matter, what could be the permission to use the data:

The license field on `.geojson` files is set to osm forums: https://community.openstreetmap.org/t/gdal-tukee-maastotietokannan-mtk-gml-tiedostomuotoa/64497/38

Screenshot of relevant parts in there:

![Image](https://github.com/user-attachments/assets/b8e38a1b-809c-4776-943d-ac7197c7b010)

Machine translation which I fixed a bit:
> To those who have inquired about this matter earlier, we have responded that the NLS requires exactly the same kind of reference to the original source as, for example, other parties providing materials have on OSM's website http://www.openstreetmap.org/copyright (E.g., Contains data from the National Land Survey of Finland's topographic database, 2013) 
> We cannot create separate user- or product-specific licenses. We are involved in preparing a public administration recommendation for open data licensing and will replace our current own license with it. That public administration recommendation will be CC 4.0 compatible and thus perhaps the most internationally suitable license for licensing data.

\+ the email "footer"

I do not see a waiver in that email above.

---

In OSM contributors page: https://wiki.openstreetmap.org/wiki/Contributors#National_Land_Survey_of_Finland

I found this:

![Image](https://github.com/user-attachments/assets/ae2d5a30-c31f-4092-9f28-25c577da05c8)

Slightly fixed machine translation:

> Hi,
> First of all, apologies for the delay in responding again, but we had to take this to our legal team for review.
Unfortunately, the received statement from them was that we do not separately issue such statements on the application of the license. In the future, there may arise unexpected problems related to the interpretation of the license, and it would also violate the equal treatment of our data users. Our license is in accordance with the public administration recommendation
and therefore we cannot give any other statements on it.
> But we still do not see any conflict between the use of OSM and our license.

---

And then I found this issue: https://github.com/openstreetmap/openstreetmap-website/pull/2580

Which to my understanding is about giving the required attribution to NLS on OSM website.

In there is another email thread discussing the attribution contents, but interestingly on the bottom of the screenshot they at NLS mention again that "attribution must be as defined in CC-BY 4.0":

![Image](https://github.com/user-attachments/assets/b824ac83-f4fc-4918-8c88-f5247127823c)

---

According to this page: https://wiki.openstreetmap.org/wiki/Fi:Maastotietokanta

Old NLS data (before license change to CC-BY 4.0) is okay to use and import, but the current data is not.

---

### My conclusion/understanding

Old NLS data is okay to use (licensed under: https://www.maanmittauslaitos.fi/avoindata-lisenssi-versio1)

We do not have waiver for current NLS data, but the usage is continued since the consensus seems to be that NLS will never question the usage, and even if they did, the risk of consequences is low*. And to me it seems like the included email threads are there primarily to strengthen the belief that NLS will continue to be okay with the continued usage.

Because NLS does not have waiver, this causes questions about double standards, since this has similar license, but was rejected: https://github.com/osmlab/editor-layer-index/pull/1608, regardless NLS still continues to be included.

I think we just need to make sure that if NLS data is included, it has to be under that previous license, so old data only. With strict interpretation, usage of current NLS data is not okay because it is CC-BY 4.0, so that imagery should not be used without waiver, which NLS seems to be unwilling to give us. If I'm correct, it means current NLS rasters would have to be removed.

\* I personally believe that too, but legally that's probably quite a weak position and should therefore probably be questioned.

---

I read so many forum posts, github issues and wiki pages trying to figure out which is the most recent or correct source. In the process, buried inside auto-hidden section of a github issue I found this: https://wiki.openstreetmap.org/w/images/7/75/HSL_permission.pdf I did not even know that such waiver existed and that data was allowed to be used while editing. While trying to investigate this, I felt like licensing documents and permissions were all over multiple places and I had to parse forum posts and wiki pages like a historian. I just wish that as the FAQ.md says, all waivers were in this repository beautifully next to the `.geojson` files.

---

This topic is complex and resources are in two languages, so I could be totally wrong. Also, I'm not claiming to be an expert in this subject.

**Comment by simonpoole** [comment]
@jimboolio are you referring to imagery rasters, that is orthoimagery or DSM/DHM or to actual geo-data? The  discussion of the later doesn't belong here (if there is a new discussion on that somewhere a pointe would be helpful, I would have a suggestion how NLS hand could be forced on the matter).

**Comment by jimboolio** [comment]
1. I was trying to figure out if the NLS image rasters/ortophoto are okay to use? Currently that resource is clearly CC-BY 4.0 which is not ok for osm which was the question of this gh issue. Because the sources I listed on my previous comment seem to point that the osm community has just accepted that this ortophoto is okay to use, since it was available for so long and no waiver can be found*. Is this correct and should it be like this?
 2. Because according to my observation NLS imagery seems to be either forgotten or granted an license exception, I tried to figure out why or with what impacts. In other words, does the same logic apply to some other data source? It did not apply to CC-BY 4.0 Helsinki imagery. So my second question became what data can I actually use? I know that this second question does not belong here, but it is only raised because of the observed NLS raster licensing exception. That raises the following question:
     1. What data in general is already accepted for use? Is there a list somewhere? **What should I use for a source of truth?** For imagery I assume this repository, even though some waivers seem to be missing. For other helpful data it is not so clear. I assume osm wiki. If that's the case, it needs updating.

 
\* As I tried to explain on the comment above, none of the resources I found grant permission for use, they are email threads in which NLS confirms that osm is ok to use the data in their mind, but they never grant any exception for osm. If I compare those against the aspects in `Licence/Waiver and Permission Templates`, the aspects are not covered on those email threads and NLS employees are reconfirming again and again that CC-BY 4.0 applies to the data.

**Comment by ilmaisin** [comment]
The situation is basically that NLS refuses to sign a formal waiver, because they claim it would be unnecessary and redundant, as OSM is already in compliance. Also, they would consider special waivers to be in contradiction with the equal treatment of data users.

However, according to OSM, the licenses aren't necessarily compatible.

What complicates matters is that the issue isn't limited to OSM, though, but it potentially affects reusers of OSM data as well.

Aaand there is also the legal question on whether or not maps drawn according to ortophotos should be considered as derivative works in Finnish law, in the first place.

**Comment by simonpoole** [comment]
> The situation is basically that NLS refuses to sign a formal waiver, because they claim it would be unnecessary and redundant, as OSM is already in compliance. Also, they would consider special waivers to be in contradiction with the equal treatment of data users.

In a way this should be simple to resolve (just referring to data here, not ortho imagery): find somebody that uses NLS data and distributes the results with restrictions that go further than what CC BY allows (aka any non-open license), then point out to NLS that they are not enforcing their license and request them to give you that in writing that you will be treated the same.

**Comment by jimboolio** [comment]
> > The situation is basically that NLS refuses to sign a formal waiver, because they claim it would be unnecessary and redundant, as OSM is already in compliance. Also, they would consider special waivers to be in contradiction with the equal treatment of data users.
> 
> In a way this should be simple to resolve (just referring to data here, not ortho imagery): find somebody that uses NLS data and distributes the results with restrictions that go further than what CC BY allows (aka any non-open license), then point out to NLS that they are not enforcing their license and request them to give you that in writing that you will be treated the same.

Sure, with data that makes sense. So with NLS _data_ this needs to happen before data can be used directly.

---

Coming back to ortophotos/rasters.. 

> The situation is basically that NLS refuses to sign a formal waiver, because they claim it would be unnecessary and redundant, as OSM is already in compliance. Also, they would consider special waivers to be in contradiction with the equal treatment of data users.

Does logic from above apply here? We have written proof (multiple copies of email threads) that NLS does not enforce their license with regards of OSM using their rasters for tracing. In other words, NLS sees OSM already in compliance.

>   * use of CC BY imagery for tracing: the OSMF has typically for good form asked for a waiver of any rights in the traced data, but even if that isn't forthcoming it would be a bit of an uphill battle for the source to claim infringement (note this is limited to actual  imagery, lidar data and similar, not to images of data). With other words this is in general unproblematic.

**Am I correct with this?:**
"We are okay to use NLS CC-BY 4.0 rasters without a waiver in OSM because we have written proof that NLS does not enforce their license with regards of OSM usage. And that backed up by the fact it would be hard to enforce it afterwards even if they changed their mind, because proving that specifically their raster was traced is difficult."

**Comment by ilmaisin** [comment]
Per my communication with them, I find it exceedingly unlikely that NLSFI would ever actually go after OSM, but what are the consequences of having CC-BY "tainted" data in OSM? Can it cause any substantial problems for data reuse, for example?

The earlier argument that the PSI directive would somehow invalidate the CC-BY restrictions is really suspicious, since the _Laki viranomaisten toiminnan julkisuudesta_, to which the directive is transposed to, does not mention copyrights at all, and IMO feels bit condescending, when it comes from someone who doesn't understand Finnish.

However, what would be worth checking out is that how far-reaching map copyrights are in Finnish statutory and case law.
